backend/src/main/java/com/google/sps/GoogleClassFactory.java

```diff
@@ -21,7 +22,7 @@
 import java.security.GeneralSecurityException;
 import java.util.List;
 
-public class GmailFactory {
+public class GoogleClassFactory {
```

Comment:
Using a factory for this object is a good idea. I'd recommend splitting it into a different factory class and keeping this class for gmail. 

On larger projects, code is (ideally) organized by feature. One should be able to exclude a feature (and all of its dependencies) from the application just by removing the package that supports the feature. Imagine if Ad-Lib grew to use many other external APIs that were constructed in a single GoogleClassFactory; even if one wanted to turn the calendar feature off by removing it from the build, this class would still exist and pull in all the calendar dependencies.